lib/pathway/rspec/matchers/accept_optional_fields.rb

```diff
@@ -6,27 +6,40 @@
   match do |form|
     @form, @fields = form, fields
 
-    not_defined.empty? && not_optional.empty?
+    not_defined.empty? &&
+      not_optional.empty? &&
+      allowing_null_values_matches? &&
+      not_allowing_null_values_matches?
   end
 
   match_when_negated do |form|
+    raise NotImplementedError, 'expect().not_to accept_optional_fields.not_allowing_null_values is not supported.' if @not_allowing_null_values
+
     @form, @fields = form, fields
 
-    not_defined.empty? && optional.empty?
+    not_defined.empty? &&
+      optional.empty? &&
+      allowing_null_values_matches? &&
+      not_allowing_null_values_matches?
   end
 
   description do
-    "accept #{field_list} as optional #{pluralize_fields}"
+    null_value_allowed = @allowing_null_values ? ' allowing null values' : ''
+    null_value_disallowed = @not_allowing_null_values ? ' not allowing null values' : ''
+    "accept #{field_list} as optional #{pluralize_fields}#{null_value_allowed}#{null_value_disallowed}"
   end
 
   failure_message do
-    "Expected to accept #{field_list} as optional #{pluralize_fields} but " +
-      [not_optional_list, not_defined_list].compact.join("; and ")
+    null_value_allowed = @allowing_null_values ? ' allowing null values' : ''
+    null_value_disallowed = @not_allowing_null_values ? ' not allowing null values' : ''
+
+    "Expected to accept #{field_list} as optional #{pluralize_fields} #{null_value_allowed}#{null_value_disallowed} but " +
```

Comment:
`"#{pluralize_fields} #{null_value_allowed}"` that space between the variables is wrong.